src/storage.js

```diff
@@ -0,0 +1,93 @@
+/**
+ * Copyright (c) Hathor Labs and its affiliates.
+ *
+ * This source code is licensed under the MIT license found in the
+ * LICENSE file in the root directory of this source tree.
+ */
+
+/**
+ * Interface to use the storage object set by the client
+ *
+ * This storage should have the same methods as localStorage (from the browser) and has
+ * to support storaing js objects. It should handle any serialization needed.
+ *
+ * @class
+ * @name Storage
+ */
+class Storage {
+  constructor() {
+    if (!Storage.instance) {
```

Comment:
I don't feel it should be a singleton.

Reply:
We use a singleton to make sure the same store is used across the lib, just like WebSocketHandler.

Reply:
You already do it exporting the instance instead of the class, don't you?

Reply:
You're correct. Change it.